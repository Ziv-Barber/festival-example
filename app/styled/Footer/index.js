// @flow

import * as React from 'react' // eslint-disable-line no-unused-vars
import { Link } from 'react-router-dom'
import styled from 'styled-components'

// Other components that we need:
import { ItemsList } from '../../components/ItemsList'

// Material-ui stuff:
import MuiButton from '@material-ui/core/Button'

// Content that we need:
import footerInfo from '../../content/footer'

const FooterMenuItem = (props: { name: string, path: string }) => (
  <Link className="link" to={props.path}>
    <MuiButton>{props.name}</MuiButton>
  </Link>
)

// Create the list of items for the Lineup component:
const footerItems = footerInfo.map((value) => ({
  key: value.key,
  component: FooterMenuItem,
  compProps: {
    name: value.name,
    path: value.path
  }
  // gridProps
}))

/**
 * footer react component.
 */
export default styled(ItemsList).attrs({
  items: footerItems,
  caption: '',
  captionProps: '',
  captionVariant: '',
  containerId: '',
  gridRootDefProps: {
    spacing: 8,
    justify: 'center',
    alignItems: 'center'
  },
  gridDefProps: {
    xs: 2,
    sm: 2,
    md: 2,
    lg: 1,
    xl: 1
  }
})`
  margin: 0 !important;
  padding: ${(props) => props.theme.spacing.unit}px 0 !important;
  width: 100% !important;
  background: #001433 !important;
  box-shadow: 0px -2px 5px 0px #fff !important;
  color: white;

  & .listGrid {
    margin: 0 !important;
    width: 100% !important;
    padding: ${(props) => props.theme.spacing.unit}px !important;
  }

  & .listItem {
    margin: ${(props) => props.theme.spacing.unit}px !important;
    z-index: 1050;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  & .link {
    text-decoration: none;
  }
`
