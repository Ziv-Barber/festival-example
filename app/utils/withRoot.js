// @flow

import * as React from 'react' // eslint-disable-line no-unused-vars
import { MuiThemeProvider } from '@material-ui/core/styles'
import { ThemeProvider } from 'styled-components'
import CssBaseline from '@material-ui/core/CssBaseline'
import theme from '../theme'

/**
 * withRoot HOC to wrap a react component with the app's theme provider.
 */
const withRoot = (Component: React.ComponentType<any>) => {
  const withRootBase = (props: any) => (
    <MuiThemeProvider theme={theme}>
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <CssBaseline />
          <Component {...props} />
        </React.Fragment>
      </ThemeProvider>
    </MuiThemeProvider>
  )

  // $FlowFixMe
  withRootBase.displayName = `withRoot(${Component.displayName ||
    Component.name})`

  return withRootBase
}

export default withRoot
