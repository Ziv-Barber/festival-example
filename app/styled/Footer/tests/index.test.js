// @flow

import React from 'react' // eslint-disable-line no-unused-vars
import { render, cleanup } from 'react-testing-library'
import { BrowserRouter as Router } from 'react-router-dom'

import Footer from '../index'
import withRoot from '../../../utils/withRoot'

// For flow:
let { describe, it, afterEach, expect } = global

const TestComp = (prop) => (
  <Router>
    <Footer />
  </Router>
)

/** @test {Footer} */
describe('<Footer />', () => {
  afterEach(cleanup)

  /** @test {Footer} */
  it('should render', () => {
    const App = withRoot(TestComp)
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })
})
