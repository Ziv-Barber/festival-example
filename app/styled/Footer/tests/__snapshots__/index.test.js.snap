// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render 1`] = `
<div>
  <div
    class="MuiPaper-root-1 MuiPaper-elevation0-3 Footer-sc-15rsv6x-0 kQiiTI"
  >
    <div
      class="MuiGrid-container-28 MuiGrid-spacing-xs-8-50 MuiGrid-align-items-xs-center-36 MuiGrid-justify-xs-center-45 listGrid"
    >
      <div
        class="MuiGrid-item-29 MuiGrid-grid-xs-2-58 MuiGrid-grid-sm-2-72 MuiGrid-grid-md-2-86 MuiGrid-grid-lg-1-99 MuiGrid-grid-xl-1-113 listItem"
      >
        <a
          class="link"
          href="/newsletter"
        >
          <button
            class="MuiButtonBase-root-151 MuiButton-root-125 MuiButton-text-127 MuiButton-flat-130"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label-126"
            >
              Newsletter
            </span>
            <span
              class="MuiTouchRipple-root-154"
            />
          </button>
        </a>
      </div>
      <div
        class="MuiGrid-item-29 MuiGrid-grid-xs-2-58 MuiGrid-grid-sm-2-72 MuiGrid-grid-md-2-86 MuiGrid-grid-lg-1-99 MuiGrid-grid-xl-1-113 listItem"
      >
        <a
          class="link"
          href="/press"
        >
          <button
            class="MuiButtonBase-root-151 MuiButton-root-125 MuiButton-text-127 MuiButton-flat-130"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label-126"
            >
              Press
            </span>
            <span
              class="MuiTouchRipple-root-154"
            />
          </button>
        </a>
      </div>
      <div
        class="MuiGrid-item-29 MuiGrid-grid-xs-2-58 MuiGrid-grid-sm-2-72 MuiGrid-grid-md-2-86 MuiGrid-grid-lg-1-99 MuiGrid-grid-xl-1-113 listItem"
      >
        <a
          class="link"
          href="/privacy"
        >
          <button
            class="MuiButtonBase-root-151 MuiButton-root-125 MuiButton-text-127 MuiButton-flat-130"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label-126"
            >
              Privacy
            </span>
            <span
              class="MuiTouchRipple-root-154"
            />
          </button>
        </a>
      </div>
      <div
        class="MuiGrid-item-29 MuiGrid-grid-xs-2-58 MuiGrid-grid-sm-2-72 MuiGrid-grid-md-2-86 MuiGrid-grid-lg-1-99 MuiGrid-grid-xl-1-113 listItem"
      >
        <a
          class="link"
          href="/contact"
        >
          <button
            class="MuiButtonBase-root-151 MuiButton-root-125 MuiButton-text-127 MuiButton-flat-130"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label-126"
            >
              Contact
            </span>
            <span
              class="MuiTouchRipple-root-154"
            />
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;
